elixir/lib/mix/tasks/generate.ex

```diff
@@ -359,6 +379,53 @@ defmodule Mix.Tasks.Contracts.Generate do
     "object" not in types_list
   end
 
+  @doc """
+    Generates the cloud events conversion functions for the contract
+  """
+  def trento_cloudevent_conversion_functions() do
+    quote do
+      @doc """
+        Serialize and return a module struct from a cloud event json
+      """
+      def serialize_from_cloud_event(json_cloud_event) do
+        with {:ok, event} <- Cloudevents.from_json(json_cloud_event) do
+          create_contract_from_cloud_event(event)
+        end
+      end
+
+      @doc """
+        Serialize and return a cloud event json from a contract struct
+      """
+      def serialize_to_cloud_event(contract) do
```

Comment:
What is the signature in this case? Does it receive a binary or some already predifined struct?

Reply:
It receives the __MODULE__ the struct of the contract we are generating, I will update them with the signatures